src/client/app_configuration_http.rs

```diff
@@ -47,8 +49,8 @@ impl AppConfigurationClientHttp<LiveConfigurationImpl> {
         configuration_id: ConfigurationId,
         offline_mode: OfflineMode,
     ) -> Result<Self> {
-        let server_client =
-            ServerClientImpl::new(service_address.clone(), token_provider.dyn_clone())?;
```

Comment:
Using `Arc` instead of our dynamic-clone. This way we share the same `TokenProvider`